### DISCUSSION TITLE
Chromium Dev Tools / Inspect

### DISCUSSION THREAD

**Discussion by Draracos** [guysoft/FullPageOS#590] [Q&A] [answered]
Hello,
We utilize FullPageOS to display dashboards in my company. We have about 300 PIs using FullpageOS for this purpose, and they work great, most of the time. What I need is another way to troubleshoot these devices to pin down what an issue is if the PI is failing to work. Sometimes its a network problem, sometimes the SD card is corrupted, and sometimes it is just on the wrong network VLAN. I can fix these.

But there are times when we are developing a new dashboard and we need to see the Inspect tab, the Chrome Dev Tools, where you can inspect the page, and see what is being called, and what is failing. The Dashboards work fine on the a regular computer, and usually even work on a regular PI that lets us inspect the tabs properly, but we have issues on FullPageOS sometimes, and having access to this Inspect tool would provide helpful knowledge to let us better troubleshoot and fix the issues. When trying to inspect, none of the shortcuts work, and all the menus have the Developer Tools greyed out.

Any assistance would be greatly appreciated. If anyone has a way to open these tools in FullPageOS, or if there is some way that we need to install it separately, Please let me know. It would be much more efficient than having our Developers spend countless hours throwing random stuff at it to try and make something stick. Lol.

![image](https://github.com/user-attachments/assets/38103be9-a7a4-4612-8860-9028b7bbdb24)

**Comment by guysoft** [comment]
Hey @Draracos
If you do ctrl+t on the FullPagOS instance it should open a new window with a tab that is not in kiosk mode, and then you can open the developer tools from there.

I am not sure if this is the flow you want but its a way to see it on FullPageoS.

Another solution I never tried:
Also, you can change the command that starts chromium and you can have the developer tools:
https://stackoverflow.com/a/27454778/311268
Says here: 
```
Add "--remote-debugging-port=9222" to /etc/chrome_dev.conf
```
I didn't try that, but you should be able to start remote debugging on chromium.

Also, since you have 300 pis, and it looks like its helping your business, I would warmly recommend making a donation to the project, its only me maintaining it out of my free time, no one pays me to do it. I have a non-related full time job. Its 100% free software. There is a donation button here: 
https://github.com/guysoft/FullPageOS?tab=readme-ov-file#donate

**Reply by guysoft** [reply]
@Draracos did it work for you in the end?

**Reply by Draracos** [reply]
Doesn't look like it, opening up a second tab with ctrl+t still shows the developer tools greyed out. And attempting to follow the instructions in that link you sent didn't get me anywhere. I should mention I am trying to figure this out on an older version since the majority of our devices are all running this version from 2022, version 0.13.0. Figuring it out for this version would prove much more effective than using a newer version and then having to reimage so many PIs that are in use. Regardless, we did find a solution for the couple Issues we were having that we needed the Dev Tools for, we were having an issue with certificates it seems. We were able to fix it by modifying the start_chromium_browser script and add "--ignore-certificate-errors" to the script after "chromium-browser".

**Reply by Draracos** [reply]
Once again this was solved by our Dev team throwing endless hours at it trying as many ideas as possible before something stuck and worked. So still figuring out a way to see a live status for the page that is failing to load would be very helpful. Just at a loss as to how to do so.

**Reply by guysoft** [reply]
Cool, sorry tmy suggestion didnt work.
Do you want to share the solution?

**Reply by Draracos** [reply, marked as answer]
Once we figured it out, it turned out to be quite simple of a fix for our issue, but again, this doesn't fix our need for a Debugging Tool.

The fix we found is specifically for Certificate Errors causing the site to not load on FullpageOS. You can do this physically on the PI by opening another terminal with ctrl+alt+f2 or f3. In the filesystem of the PI, sudo nano ./scripts/start_chromium_browser to open the starting script for chromium. Then edit the **# Standard behavior - runs chrome** section to include _--ignore-certificate-errors_ like below

```
# Standard behavior - runs chrome
chromium-browser --ignore-certificate-errors --kiosk --touch-events=enabled --disable-pinch --noerrdialogs --disable-session-crashed-bubble --simulate-outdated-no-au='Tue, 31 Dec 2099 23:59:59 GMT' --disable-component-update --overscroll-history-navigation=0 --disable-features=Translate --app=$(/home/pi/scripts/get_url)
exit;
```
